backend/core/src/main/java/org/sonarsource/sonarlint/core/rules/RuleDetailsAdapter.java

```diff
@@ -78,16 +80,15 @@ private RuleDetailsAdapter() {
   }
 
   public static EffectiveRuleDetailsDto transform(RuleDetails ruleDetails, @Nullable String contextKey) {
-    org.sonarsource.sonarlint.core.commons.IssueSeverity defaultSeverity = ruleDetails.getDefaultSeverity();
-    RuleType type = ruleDetails.getType();
+    var cleanCodeAttribute = ruleDetails.getCleanCodeAttribute().map(RuleDetailsAdapter::adapt).orElse(null);
+    Either<StandardModeDetails, MQRModeDetails> severityDetails = cleanCodeAttribute != null && !ruleDetails.getDefaultImpacts().isEmpty() ?
+      Either.forRight(new MQRModeDetails(cleanCodeAttribute, Objects.requireNonNull(toDto(ruleDetails.getDefaultImpacts())))) :
```

Comment:
you can remove the requireNonNull now that the list can't be null `Objects.requireNonNull(toDto(ruleDetails.getDefaultImpacts())))`